### PR TITLE
Issue #1411: Fix gitignore applying rules to files in subdirectories

### PR DESCRIPTION
**Pull request by oxyc**
Not entirely sure when this was added to git but it's definitely been in for a while. Somewhere between ~1.9.3 and 2.2.1~

### CODE REVIEW

**Comment by oxyc** [comment]
Found the commit message https://github.com/git/git/commit/f87f94974869b1ddd579d4c6aaff12a5f86ca26e. Seems to have been in since forever.

File: .gitignore [hunk 1]
```diff
@@ -15,15 +15,12 @@
...
-drupal.make.yml
-Vagrantfile.local
-examples/prod/inventory
-examples/prod/bootstrap/vars.yml
-scripts/
-drupal/
-
-# Some roles include 'config.yml' files.
-!provisioning/roles/**/config.yml
+/local.config.yml
```
Comment (geerlingguy) [inline .gitignore]: Since `local.config.yml` should only be used for a local instance configuration... maybe we leave that as globally ignored?

Comment (oxyc) [inline .gitignore]: Fixed, and `Vagrantfile.local` as well.

**Comment by geerlingguy** [comment]
Since this has the potential to disrupt some projects (since previously-ignored files could be inadvertently committed), I think I may wait until after the next stable release to merge this. Not major enough to warrant a full version bump or anything, but I _am_ thinking it might be best to wait until the next minor version...

**Comment by geerlingguy** [comment]
I might merge this in and bump the minor version since there's now a dozen or so smallish changes... what do you think?

**Comment by oxyc** [comment]
Sounds good. We don't really have any PRs or issues that would warrant a minor version bump so no need to wait I guess. Except for the composer one and that's for 5.0.0

**Comment by oxyc** [comment]
Also the automagic installation of vagrant plugins might justify a minor bump.

**Comment by geerlingguy** [comment]
Yeah, I'm thinking the next release will be a minor bump—a lot of small things that together could cause at least one breaking change in more complex scenarios.

**Pull request merged**